### ISSUE TITLE
Conditional attributes bug??

### ISSUE THREAD

**Issue by Cakirucho** [Netflix/fast_jsonapi#284]
Hi guys, first of all i wanted to thx you for the great library you are building.

I have been trying the new `Conditional attributes` feature, and it seems that there is one case where it do not works:

In the Documentation, you are using Proc like that:
```ruby
attribute :release_year, if: Proc.new do |record|
  # Release year will only be serialized if it's greater than 1990
  record.release_year > 1990
end
```

But, as you can't build `Proc` object without block, that piece of code is raising `tried to create Proc object without a block` exception.

I have been able to use Conditionals with custom attributes by using it like that:

```ruby
attribute :my_best_attribute, if: Proc.new {
    |record, params| params.present? && params[:true_condition] == true
  } do |record, params|
    record.custom_happy_method()
end
```

I am doing something wrong, it is a bug?? or it is that the Documentation is not well documented in that part?.

Thx.

PD: I have been taking a look in your tests and it seems that there is only tests for Model attributes (and not those for custom serializer attributes).

```ruby
class MovieOptionalRecordDataSerializer
    include FastJsonapi::ObjectSerializer
    set_type :movie
    attributes :name
    attribute :release_year, if: Proc.new { |record| record.release_year >= 2000 }
end

class MovieOptionalParamsDataSerializer
    include FastJsonapi::ObjectSerializer
    set_type :movie
    attributes :name
    attribute :director, if: Proc.new { |record, params| params && params[:admin] == true }
end

class MovieOptionalRelationshipSerializer
    include FastJsonapi::ObjectSerializer
    set_type :movie
    attributes :name
    has_many :actors, if: Proc.new { |record| record.actors.any? }
end

 class MovieOptionalRelationshipWithParamsSerializer
    include FastJsonapi::ObjectSerializer
    set_type :movie
    attributes :name
    belongs_to :owner, record_type: :user, if: Proc.new { |record, params| params && params[:admin] == true }
end
```

**Comment by TrevorHinesley** [comment]
@Cakirucho To your first point, you're correct: it should use brackets and not the `do` syntax. That's my bad!

Referring to your last comment: please add tests if you feel so inclined!

**Comment by Cakirucho** [comment]
@TrevorHinesley thx for that Trevor!!

I will take a look on test and try to add couple of them for custom attributes :)

**Comment by shishirmk** [comment]
Released in [Version 1.4](https://github.com/Netflix/fast_jsonapi/releases/tag/1.4)
